### PR TITLE
34.4

### PR DESCRIPTION
**Pull request by yutsevich**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+public class CinemaHallDaoImpl implements CinemaHallDao {
+
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(cinemaHall);
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java:23]: persist?

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java:23]: persist?

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java:44]: the same - persist?

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Table(name = "movie_sessions")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/model/MovieSession.java:23]: Fetch type Eager by default. Shouldn't we use Lazy? If so - you need to add in findAll... method this line:
root.fetch("movie");

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    private Movie movie;
+    @ManyToOne
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/model/MovieSession.java:25]: as above...

**Review by fyarkiy — APPROVED** [review]
Great!!!

File: src/main/java/com/dev/cinema/dao/MovieDao.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.dev.cinema.dao;
+
+import com.dev.cinema.model.Movie;
+import java.util.List;
+
+public interface MovieDao {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/MovieDao.java:6]: цієї дз тут не повинно бути

**Pull request merged**